scripts/src/java/org/oppia/android/scripts/RegexPatternValidationCheck.kt

```diff
@@ -0,0 +1,232 @@
+          filenameCheckBuilder.mergeFrom(it)
+        }.build() as FilenameChecks
+
+      return namePatternsObj.getFilenameChecksList()
+    }
+
+    /**
+     * Fetches all file content checks
+     *
+     * @return [List<FileContentCheck>] a list of all the FileContentChecks
+     */
+    fun getFileContentChecks(): List<FileContentCheck> {
+      val fileContentsBinaryFile =
+        File("scripts/assets/file_content_validation_checks.pb")
+      val fileContentCheckBuilder = FileContentChecks.newBuilder()
+      val fileContentsObj: FileContentChecks =
+        FileInputStream(fileContentsBinaryFile).use {
+          fileContentCheckBuilder.mergeFrom(it)
+        }.build() as FileContentChecks
+
+      return fileContentsObj.getFileContentChecksList()
+    }
+
+    /**
+     * Checks for a prohibited file naming pattern
+     *
+     * @param repoPath the path of the repo.
+     * @param searchFiles a list of all the files which needs to be checked.
+     * @param fileNameRegexString filename pattern regex which should not be present
+     * @param errorToShow error to show incase of failure
+     * @return [Boolean] check failed or passed
+     */
+    fun checkProhibitedFileNamePattern(
+      repoPath: String,
+      searchFiles: Sequence<File>,
+      prohibitedFilenameRegexString: String,
+      errorToShow: String
+    ): Boolean {
+      var filenamePatternCheckFailedFlag = false
+      val prohibitedFilenameRegex = prohibitedFilenameRegexString.toRegex()
+      searchFiles.forEach {
+        val filePath = it.toString().removePrefix(repoPath)
+        if (prohibitedFilenameRegex.matches(filePath)) {
+          logProhibitedFilenameFailure(
+            filePath = filePath,
+            errorToShow = errorToShow
+          )
+          filenamePatternCheckFailedFlag = true
+        }
+      }
+      return filenamePatternCheckFailedFlag
+    }
+
+    /**
+     * Checks for a prohibited file content
+     *
+     * @param repoPath the path of the repo.
+     * @param searchFiles a list of all the files which needs to be checked.
+     * @param fileNameRegexString filename regex string in which to do the content check
+     * @param prohibitedContentRegexString regex string which should not be contained in the file
+     * @param errorToShow error to show incase of failure
+     * @return [Boolean] check failed or passed
+     */
+    fun checkProhibitedContent(
+      repoPath: String,
+      searchFiles: Sequence<File>,
+      fileNameRegexString: String,
+      prohibitedContentRegexString: String,
+      errorToShow: String
+    ): Boolean {
+      var contentCheckFailedFlag = false
+      val fileNameRegex = fileNameRegexString.toRegex()
+      val prohibitedContentRegex = prohibitedContentRegexString.toRegex()
+      searchFiles.forEach {
+        if (fileNameRegex.matches(it.name)) {
+          var lineNumber = 0
+          File(it.toString()).forEachLine { lineString ->
+            lineNumber++
```

Comment:
This is another bit of logic that would be nice to extract into a helper class like RepoFile:

```kotlin
private class RepoFile(...) {
  /** ... */
  fun enumerateLines(callback: (String, Int) -> Unit) { ... }
}
```

Also, if you'd like to avoid the var here you could do something like:

```kotlin
File(it.toString()).bufferedReader().lineSequence().foldIndexed ....
```

Since [``foldIndexed``](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.sequences/fold-indexed.html) provides the line index within the reader, and lets you avoid the ``contentCheckFailedFlag`` variable. That being said, the callback approach above would probably be simpler (and could be done by via a forEachIndexed call on the line sequence of the file to avoid a lineNumber var).

Reply:
I would really love to go by the `foldIndexed` approach. My primary reason for choosing this is that, for the new contributors, this approach is easier to understand. I myself was facing difficulty in the first look of understanding the first approach.